### PR TITLE
Retry db connection on failure

### PR DESCRIPTION
**Pull request by benjefferies**
Retry database connection on DatabaseError every 2s for up to 30s. This
allows the app to be more robust when starting up. For instance when
running up apps via docker-compose the app should be able to recover if
the database is not up first.

### CODE REVIEW

**Review by tej99 — APPROVED** [review]
LGTM. Out of curiosity, what's the advantage of using the retrying package instead of a for loop with time delay?

**Comment by benjefferies** [comment]
@tej99 Less code, the code has already been tested

File: Pipfile [hunk 1]
```diff
@@ -24,13 +27,14 @@ requests = "==2.17.3"
...
 ras-common-utils = {git = "https://github.com/ONSdigital/ras-common-utils.git", ref = "0.0.7"}
+retrying = "*"
 
 
 [dev-packages]
 
 tox = "*"
 coverage = "*"
-flake8 = "*"
+"flake8" = "*"
```
Comment (JamesGardiner) [inline Pipfile:37]: why the quotes?

File: run.py [hunk 2]
```diff
@@ -30,6 +32,11 @@ def create_app(config):
     return app
 
 
+def retry_if_database_error(exception):
+    return isinstance(exception, DatabaseError)
+
+
+@retry(retry_on_exception=retry_if_database_error, wait_fixed=2000, stop_max_delay=30000)
```
Comment (JamesGardiner) [inline run.py]: Do you want to gracefully handle the `RetryError` that is raised?

Comment (JamesGardiner) [reply run.py]: https://github.com/rholder/retrying/blob/1d5699348d707e377aad7488da6a8a1b48a65933/retrying.py#L286

File: Pipfile (hunk 1 above)
Comment (benjefferies) [reply Pipfile:37]: It was automatically added when pipenv install. I can remove

File: run.py (hunk 2 above)
Comment (benjefferies) [reply run.py]: I guess the only way we can handle it is by logging the error as at that point it seems irrecoverable?

File: Pipfile (hunk 1 above)
Comment (jcox-dev) [reply Pipfile:37]: Leave it in. It's Pipenv normalising the package names. Those with numbers or special characters are intentionally quoted probably from checking against PyPI real names.

**Review by jcox-dev — APPROVED** [review]

**Review by JamesGardiner — APPROVED** [review]

**Pull request merged**